google/src/main/java/cz/ackee/rxsociallogin/google/RxGoogleLogin.kt

```diff
@@ -19,20 +19,20 @@ import io.reactivex.SingleEmitter
 /**
  * Class that handles Google login logic and transforms the result into reactive response.
  */
-class RxGoogleLogin(context: Context, additionalScopes: Array<Scope> = arrayOf()) {
+class RxGoogleLogin(
+    context: Context,
+    additionalScopes: Array<Scope> = arrayOf(),
+    private val tokenType: GoogleTokenType = GoogleTokenType.SERVER_AUTH_CODE) {
 
     companion object {
         private const val REQUEST_SIGN_IN = 10000
     }
 
-    private val googleSignInClient = GoogleSignIn.getClient(context, GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
-            .requestServerAuthCode(context.getString(R.string.google_server_client_id).also {
-                if (it.isEmpty()) {
-                    throw RuntimeException("Valid google_server_client_id should be provided in the app")
-                }
-            })
-            .requestScopes(Scope(Scopes.EMAIL), *additionalScopes)
-            .build())
+    private val client = GoogleSignIn.getClient(context, GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
+        .requestToken(context.getString(R.string.google_server_client_id))
```

Comment:
No pokud vim tohle ted' prestane fungovat protoze jsi odstranil `requestServerAuthCode`. Melo by to zadat o kod na zaklade toho typu

Reply:
Jo sorry uz vidim